### PR TITLE
burpsuite: 2023.7.2 -> 2023.10.2.4, add Professional Edition

### PR DESCRIPTION
**Pull request by arcayr**
## Description of changes
Bump Burp Suite from 2023.7.2 to 2023.10.2.4, and adds the ability to install the Professional Edition of Burp Suite by overriding the `proEdition` argument to `true`.

Some of upstream's URLs now redirect from the word "free" to "community". They have been amended as part of this update.

A list of upstream changes since the last version in nixpkgs, from [https://portswigger.net/burp/releases](https://portswigger.net/burp/releases):
* 2023.10.2.4: We have upgraded Burp's built-in browser to 118.0.5993.117 for Mac / Linux and 118.0.5993.90 for Windows. For more information, see the [Chromium release notes](https://chromereleases.googleblog.com/2023/10/stable-channel-update-for-desktop_24.html).
* 2023.10.2.3: We have upgraded Burp's built-in browser to 118.0.5993.88 for Mac / Linux and 118.0.5993.88/.89 for Windows. This update contains a security fix. For more information, see the [Chromium release notes](https://chromereleases.googleblog.com/2023/10/stable-channel-update-for-desktop_17.html).
* 2023.10.2.2:  This release introduces new functionality for BChecks, including the ability to test your checks from within the editor and create definitions from a blank template. We have also added a notes feature to Repeater tabs. 
* 2023.10.1.2: We have upgraded Burp's built-in browser to 117.0.5938.132 for Mac, Linux, and Windows. This update contains security fixes. For more information, see the [Chromium release notes](https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_27.html).
* 2023.10.1.1: This release includes the introduction of user activity logging, and a number of other improvements.
* 2023.9.4: We have upgraded Burp's built-in browser to [116.0.5845.140 for Mac and Linux and 116.0.5845.140/.141 for Windows](https://chromereleases.googleblog.com/2023/08/stable-channel-update-for-desktop_29.html). This update contains  security fixes.
* 2023.9.3: This release introduces the ability to unpack Brotli-compressed messages in the Proxy and Repeater tools, and adds Organizer functionality to the Montoya API.
* 2023.9.2: This release upgrades Burp's built-in browser and fixes a bug when scanning [GraphQL](https://portswigger.net/web-security/graphql) APIs.
* 2023.9.1: This release introduces new Repeater functionality based on the techniques discussed in James Kettle's talk "Smashing the State Machine: The True Potential of Web Race Conditions", first presented at Black Hat USA 2023. Repeater's new single-packet attack feature nullifies network jitter, enabling you to send multiple requests in parallel. These requests are synchronized to arrive within a very small time window, making it much simpler to test for race conditions.
* 2023.8.1: We have upgraded Burp's built-in browser to [115.0.5790.170 for Mac and Linux and 115.0.5790.170/.171 for Windows](https://chromereleases.googleblog.com/2023/08/stable-channel-update-for-desktop.html). This update contains multiple high-severity security fixes.
* 2023.7.3: We have upgraded Burp's built-in browser to [115.0.5790.170 for Mac and Linux and 115.0.5790.170/.171 for Windows](https://chromereleases.googleblog.com/2023/08/stable-channel-update-for-desktop.html). This update contains multiple high-severity security fixes.

## Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>burpsuite</li>
  </ul>
</details>

### CODE REVIEW

**Comment by AkechiShiro** [comment]
@Arcayr I believe that [Playwright](https://playwright.dev/) has some kind of embedded browser to be used as well and is packaged in Nixpkgs but I'm pretty sure it's not the same case as Burpsuite (it's not an embeded browser inside a .jar) https://github.com/NixOS/nixpkgs/blob/e87fc0b4720d15a847137e5e4665c55309eaddff/pkgs/development/web/playwright/driver.nix#L102

But, it might be helpful to give some ideas maybe.

**Comment by buckley310** [comment]
The embedded browser works for me. Has ever since https://github.com/NixOS/nixpkgs/pull/210155

**Comment by buckley310** [comment]
Does not appear to be weapped in any way. burp runs in a FHSUserEnv which provides missing libraries.

You can test out my working setup with `nixos-rebuild build-vm --flake github:buckley310/nixos-config#testbox`
(The default password is just the username)

image

**Comment by AkechiShiro** [comment]
On my side, I'm on ArchLinux whether on X11 or Wayland, Burpsuite installed via nix never launches, I always get an error at startup, says : 

```
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true # 2 options I've probably set as an ENV variable for text antialiasing.
Could not start Burp: java.lang.Error: Cannot load com.sun.java.swing.plaf.gtk.GTKLookAndFeel
```
If you have any idea to what this could be related @Arcayr, I will try to reproduce under NixOS using Sway in a VM but I haven't had the time to experiment yet.

**Comment by AkechiShiro** [comment]
I fear that you are right @Arcayr I'm not sure exactly how to go around debugging this kinda of issue, maybe the Java installed on Arch is conflicting with the one used by BurpSuite

**Comment by buckley310** [comment]
I have this PR a test and it works well.

I ran burp, launched the integrated browser, captured a bunch of traffic and tested some items in the repeater.

**Comment by AkechiShiro** [comment]
Following, the NixCon 2023, using `nix-shell -p --pure burpsuite`, makes BurpSuite work on any Linux distro using Nix, I believe I should document this workaround, should I create another PR ? @acairncross @RaitoBezarius

**Comment by RaitoBezarius** [comment]
This can be documented in the NixOS wiki, but I am not sure if it has any place in nixpkgs.

More generally, this seems to hint at the fact if more purity is necessary in the wrapper or there's a polluting external environment in the non-NixOS system.

It is hardly actionable on our side unless the pollution in question can be isolated.

**Comment by AkechiShiro** [comment]
I will see if there is a way I can find out what is polluting the *pureness* of the impure `nix-shell` narrowing this down, I'll then try and PR a fix.

**Comment by Artturin** [comment]
Dropped the (clean) merge commit.

**Review by Artturin — APPROVED** [review]

File: pkgs/tools/networking/burpsuite/default.nix [hunk 1]
```diff
@@ -22,20 +22,20 @@ let
     hash = product.hash;
   };
 
-  pname = "burpsuite";
+  name = "burpsuite-${version}";
```
Comment (bjornfor) [inline pkgs/tools/networking/burpsuite/default.nix:25]: Why does this version bump commit revert the pname/version change from an earlier commit in this PR?

**Review by bjornfor — APPROVED** [review]

**Comment by AkechiShiro** [comment]
Hello @Arcayr,

I would like to add auto updates to Burpsuite, I have done them in a repo for ArchLinux, do you think we could scrap that and add an `updateScript` to the Burpsuite package (it's using GitHub Actions right now but I think the bot @r-ryantm) 

Repo : https://github.com/akechishiro/aur-burpsuite
Update script : https://github.com/AkechiShiro/aur-burpsuite/blob/main/update.sh

**Pull request merged**